library/src/G4CMPBoundaryUtils.cc

```diff
@@ -60,12 +63,13 @@
 
 G4CMPBoundaryUtils::G4CMPBoundaryUtils(G4VProcess* process)
   : buVerboseLevel(G4CMPConfigManager::GetVerboseLevel()),
-    procName(process->GetProcessName()), procUtils(0),
+    procName(""), procUtils(0),
     kCarTolerance(G4GeometryTolerance::GetInstance()->GetSurfaceTolerance()),
     maximumReflections(-1), prePV(0), postPV(0), surfProp(0), matTable(0),
     electrode(0) {
+  if (process) procName = process->GetProcessName();
   procUtils = dynamic_cast<G4CMPProcessUtils*>(process);
-  if (!procUtils) {
+  if (process && !procUtils) {
```

Comment:
This could be a problem.  Is procUtils used anywhere down in the BoundaryUtils code?  If so, that would break for the case where this inherits from PhononElectrode.

Reply:
Uh, oh.  We will need to look at both of those functions, and identify either how to avoid using procUtils (though I'm guessing I put it in there for a reason!), or find a way for the electrode class to deal with it.

All this may possibly be indicating that the original idea to move the Lambertian stuff (back) into BoundaryUtils was a mistake.   If the Lambertian code was the only thing we needed to add G4CMPBoundaryUtils to the electrode, then maybe the underlying code can live in G4CMPUtils.  BoundaryUtils can have functions with the same name that call down to G4CMPUtils.

Reply:
I guess I am not understanding procUtils and why it's an issue? It was in the constructor before G4CMP-511.

Is the problem that it's in the constructor at all and that I need to call a constructor in PhononElectrode?